ledgerwallet/manifest_toml.py

```diff
@@ -0,0 +1,99 @@
+import os
+import sys
+from typing import Dict, Optional
+
+from ledgerwallet.manifest import AppManifest, icon_from_file
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import toml as tomllib
+
+from ledgerwallet import params
+
+
+class AppManifestToml(AppManifest):
+    def __init__(self, filename):
+        self.path = os.path.dirname(filename)
+        if sys.version_info >= (3, 11):
+            with open(filename, "rb") as f:
+                self.dic = tomllib.load(f)
+        else:
+            with open(filename, "r") as f:
+                self.dic = tomllib.load(f)
+
+    @property
+    def app_name(self) -> str:
+        return self.dic["name"]
+
+    def is_device_defined(self, device) -> bool:
```

Comment:
Just thinking, this PR is not the place for this, but all these `device` arguments + the similarities between most `AppManifestJson` and `AppManifestToml` methods (with only `self.dic` replaced by `self.dic[device]`) makes me think that `ledgerwallet` needs a `Device` coming out of the `AppManifest`, which would remove the format specific bits (centralized into the constructor), manage all these redundancies and offer a clean interface for higher-level code. Like (extracted from the client):
```python
        device = app_manifest.get_device(device)
        hex_file = IntelHex(device.binary)
        code_length = hex_file.maxaddr() - hex_file.minaddr() + 1
        data_length = device.data_size

        code_length -= data_length
        assert code_length % 64 == 0  # code length must be aligned

        flags = device.flags  # not handled yet

        params = device.serialize_parameters()
        main_address = hex_file.start_addr["EIP"] - hex_file.minaddr()

        if device.has_api_level:
            data = struct.pack(
                ">BIIIII",
                device.app_level,
```